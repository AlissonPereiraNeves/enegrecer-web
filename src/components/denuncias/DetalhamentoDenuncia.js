import PropTypes from 'prop-types'
import React, { Component } from 'react';
import $ from 'jquery';
import { cortarPalavra } from '../../helpers';
import { EstadoFormGroup, CampoTexto, RadioGrupoBotoes, Combobox } from '../FormGroups';
import * as ConstantesCSS from '../layouts/ConstantesCss'


export default class DetalhamentoDenuncia extends Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
    this.state = {
      detalhamento: '',
      dataOcorrencia: '',
      horaOcorrencia: '',
      idCategoria: '',
      endereco: '',
      estado: ''
    };
  }
  
  componentDidMount() {
    $(document).ready(function() {
      $('select').material_select();
    });
  }

  handleChange(value, property) {
    this.setState(
      { [property]: value },
      () => this.props.handleChange(this.state)
    );
  }

  render() {
    return (
      <div>
        <br />
        <h1>Nova Denúncia</h1>
        <br />
       

      <div className="row">
        <CampoTexto
          id={'detalhamento'}
          label={'* Detalhamento'}
          maxLen={255}
          type={'text'}
          inputClasse={ConstantesCSS.CLASSES_TEXTAREA}
          divClasse={ConstantesCSS.CLASSES_DIV_INPUT + ' col s6'}
          onChange={(e) => { this.handleChange(cortarPalavra(e.target.value, 255), 'detalhamento') }}
        />
      </div>  
      
      <div className="row">

        <RadioGrupoBotoes
        id={'idCategoria'}
        onChange={e => this.handleChange(e.target.checked, 'idCategoria')}
        botoes={[{ id: 'radioInjuria', valor: 'injuria', label: 'Injúria' },
          { id: 'radioRacismo', valor: 'racismo', label: 'Racismo' }]}
        />
         
        <CampoTexto
          id={'dataOcorrencia'}
          label={'Data do ocorrido'}
          inputClasse={ConstantesCSS.CLASSES_INPUT}
          divClasse={ConstantesCSS.CLASSES_DIV_INPUT + ' col s6' }
          maxLen={0}
          type={'date'}
          onChange={(e) => { this.handleChange(e.target.value, 'dataOcorrencia') }}
        />

        <CampoTexto
          id={'horaOcorrencia'}
          label={'Hora do ocorrido'}
          maxLen={0}
          type={'time'}
          inputClasse={ConstantesCSS.CLASSES_INPUT}
          divClasse={ConstantesCSS.CLASSES_DIV_INPUT + ' col s10' }
          onChange={(e) => { this.handleChange(e.target.value, 'horaOcorrencia') }}
        />
      </div>  

     <h4>Local do crime</h4>
      <br />

      <div className="row">
        <CampoTexto
          id={'endereco'}
          label={'Endereço'}
          maxLen={255}
          type={'text'}
          inputClasse={ConstantesCSS.CLASSES_TEXTAREA}
          divClasse={ConstantesCSS.CLASSES_DIV_INPUT + ' col s12'}
          onChange={(e) => { this.handleChange(cortarPalavra(e.target.value, 255), 'endereco') }}
        />
      </div>   
       <Combobox />
      <div className="row">
        <EstadoFormGroup
          id="estado"
          value={this.state.estado}
          divClasse={ConstantesCSS.CLASSES_DIV_INPUT + ' col s12'}
          handleChange={this.handleChange}
        />
      </div>  
      </div>
    )
  }
}

DetalhamentoDenuncia.defaultProps = { handleChange: () => { } };

DetalhamentoDenuncia.propTypes = {
  handleChange: PropTypes.func
};

