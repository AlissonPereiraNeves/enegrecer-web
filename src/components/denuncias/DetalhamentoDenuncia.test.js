import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import DetalhamentoDenuncia from './DetalhamentoDenuncia';

describe('DetalhamentoDenuncia Component', () => {
  it('deve possuir estrutura definida ', () => {
    const detalhamentoDenuncia = mount(<DetalhamentoDenuncia />);
    const detalhamentoDenunciaJson = toJson(detalhamentoDenuncia);
    expect(detalhamentoDenunciaJson).toMatchSnapshot();
  });
});

