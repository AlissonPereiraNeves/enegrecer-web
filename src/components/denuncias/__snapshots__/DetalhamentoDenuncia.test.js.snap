// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetalhamentoDenuncia Component deve possuir estrutura definida  1`] = `
<div>
  <h1>
    Nova Denúncia
  </h1>
  <div
    className="row"
  >
    <nome
      id="detalhamento"
      onChange={[Function]}
      state="detalhamento"
    />
  </div>
  <div
    className="row"
  >
    <botoesRadio
      botoes={
        Array [
          Object {
            "id": "radioInjuria",
            "label": "Injúria",
            "valor": "injuria",
          },
          Object {
            "id": "radioRacismo",
            "label": "Racismo",
            "valor": "racismo",
          },
        ]
      }
      classes="col s3"
      id=""
      state="idCategoria"
    />
    <Data
      divClasse="input-field col s4"
      id="dataOcorrencia"
      label="Data do ocorrido"
      state="dataOcorrencia"
    />
    <campoTexto
      divClasse="input-field col s5"
      id="horaOcorrencia"
      inputClasse="validate"
      label="Hora do ocorrido"
      maxLen={0}
      onChange={[Function]}
      type="time"
    />
  </div>
</div>
`;
